### ISSUE TITLE
Unable to get response after a few requests

### ISSUE THREAD

**Issue by therealjohn** [WildernessLabs/Maple.NetMF#2]
## Overview

After trying to test a RequestHandler endpoint a few times, it eventually stops working. 

## Steps to Reproduce

1. Download the [Connected Coffee](https://github.com/WildernessLabs/Netduino_Samples/tree/master/Connected_CoffeeMaker) sample app and run the ApplianceHost app on the board. 
2. Use Postman or browser to hit the `{ipaddress}/status` endpoint
3. I can usually get 3 requests off before it stops working. 

## Actual Results 

After step #3, The fourth just never responds. After that, no other request works. 

## Expected Results

Should be able to get a response from a request. 

## Additional Information

It behaves like something is deadlocking.

**Comment by therealjohn** [comment]
I suspect this might be related to the `HttpListener`. I've tried a number of debugging steps from source and nothing seems to make sense which has lead me to think it's deeper in the framework. I'll revisit again with fresh eyes.

**Comment by therealjohn** [comment]
I'm using [this](https://gist.github.com/therealjohn/6ba5cb2e55ddd1213732a6ffbbdab91e) to try and debug the basics of what Maple is doing. It does lead to an exception on the ~third request attempt. 

To reproduce:
1. I browse to `{my device ip}:8500/test` 
2. Repeat / refresh the browser a few times

Results:
```
Exception was thrown: System.NullReferenceException
System.Net.OutputNetworkStreamWrapper::Flush
System.Net.HttpListenerResponse::System.IDisposable.Dispose
System.Net.HttpListenerResponse::Finalize
```

**Comment by therealjohn** [comment]
I believe the problem I found in my last comment is this: https://netmf.codeplex.com/workitem/2158
Fixed in v4.4 of .NETMF

**Comment by bryancostanich** [comment]
good news then! we've almost got the 4.4 release ready for Netduino. :)

**Comment by hbraasch** [comment]
I think the problem lies in the routine WriteOutputStream of class RequestHandlerBase in Maple.
I copied the Maple source code and debugged it locally. I then found it stopped giving me the problem as soon as I removed the enclosing "using (this.Context.Response.OutputStream)" in WriteOutputStream. The exception trace given earlier in this thread lead me  in this direction.
Hope this helps.

**Comment by hdortim** [comment]
thanks @hbraasch, that help me, too.

**Comment by adrianstevens** [comment]
This should be fixed in Maple 0.0.10 - it's been very stable testing on a Netduino 3 WiFi
https://www.nuget.org/packages/Maple/0.0.10 

Closing issue - thx @hbraasch!
